packages/fetchai/contracts/erc1155/contract.py

```diff
@@ -52,13 +52,23 @@ def __init__(
         :param contract_interface: the contract interface.
         """
         super().__init__(contract_id, contract_config, contract_interface)
-        self.item_ids = []  # type: List[int]
+        self.token_ids = []  # type: List[int]
+        self.game_currency_id = 0  # type: int
 
-    def create_item_ids(self, token_type: int, token_ids: List[int]) -> None:
+    def generate_item_ids_based_on_nb_goods(
+        self, token_type: int, nb_goods: int
+    ) -> List[int]:
         """Populate the item_ids list."""
-        assert self.item_ids == [], "Item ids already created."
-        for token_id in token_ids:
-            self.item_ids.append(Helpers().generate_id(token_type, token_id))
+        assert self.token_ids == [], "Item ids already created."
+        for i in range(nb_goods):
+            self.token_ids.append(Helpers().generate_id(token_type, i))
+        return self.token_ids
+
+    def generate_single_item_id(self, token_type: int, nb_good: int) -> int:
+        """Create single token id"""
+        self.game_currency_id = Helpers().generate_id(nb_good, token_type)
```

Comment:
note it should be `generate_id(token_type, i)

You will have to be careful that there is no clash in token ids between this and the previous method. I think it might be better to just create both in one call..also see discussion on related file.